flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/ClusterHealthEvaluator.java

```diff
@@ -174,6 +175,25 @@ private boolean evaluateCheckpoints(
             return true;
         }
 
+        var completedCheckpointsCheckWindow =
+                configuration.get(OPERATOR_CLUSTER_HEALTH_CHECK_CHECKPOINT_PROGRESS_WINDOW);
+
+        CheckpointConfig checkpointConfig = new CheckpointConfig();
+        checkpointConfig.configure(configuration);
+        var checkpointingInterval = checkpointConfig.getCheckpointInterval();
+        var checkpointingTimeout = checkpointConfig.getCheckpointTimeout();
+        var tolerationFailureNumber = checkpointConfig.getTolerableCheckpointFailureNumber() + 1;
+        if (completedCheckpointsCheckWindow.toMillis()
+                < Math.max(
+                        checkpointingInterval * tolerationFailureNumber,
+                        checkpointingTimeout * tolerationFailureNumber)) {
+            LOG.warn(
+                    "cluster.health-check.checkpoint-progress.window is not long enough. "
+                            + "It must be greater than max(execution.checkpointing.interval * execution.checkpointing.tolerable-failed-checkpoints, "
+                            + "execution.checkpointing.timeout * execution.checkpointing.tolerable-failed-checkpoints)");
+            return true;
+        }
```

Comment:
Instead if returning a warning and turning off a feature, would it make sense to default to `2 * cpInterval` ? That is probably more useful to the user

Reply:
Looks fine to me to default to appropriate value but I'm not sure to understand why defaulting to `2 * cpInterval` instead of `Math.max(checkpointingInterval * tolerationFailureNumber,checkpointingTimeout * tolerationFailureNumber)`

Reply:
What you write makes more sense I agree :)